File: count-and-say/count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        else:
            resultString = ""
            charCount = 0
            prevChar = 0

            for pos, char in enumerate(self.countAndSay(n - 1)):
                if pos == 0:
                    charCount = 1
                    prevChar = char
                elif char == prevChar:
                    charCount += 1
                else:
                    resultString = resultString + str(charCount) + prevChar
                    charCount = 1
                    prevChar = char

            resultString = resultString + str(charCount) + prevChar

            return resultString


sol = Solution()

print(sol.countAndSay(5))

File: count-and-say/count-and-say.go
package main

import "fmt"

func countAndSay(n int) string {
	if n == 1 {
		return "1"
	} else {
		var (
			resultString string = ""
			charCount    int    = 0
			prevChar     rune   = 0
		)

		for pos, char := range countAndSay(n - 1) {
			if pos == 0 {
				charCount = 1
				prevChar = char
			} else if char == prevChar {
				charCount++
			} else {
				resultString = fmt.Sprintf("%s%d%c", resultString, charCount, prevChar)
				charCount = 1
				prevChar = char
			}
		}
		resultString = fmt.Sprintf("%s%d%c", resultString, charCount, prevChar)

		return resultString
	}
}

func main() {
	fmt.Println(countAndSay(5))
}
